### PR TITLE
Issue/361 transformation refactor

### PR DESCRIPTION
**Pull request by jonase**
- [x] Test plan | Unit test | Integration test
- [ ] Copyright header
- [x] Code formatting
- [ ] Documentation

### CODE REVIEW

File: backend/src/org/akvo/lumen/component/transformation_engine.clj [hunk 1]
```diff
@@ -11,25 +11,26 @@
...
+                     (try
+                       (when-not (identical? transformation sentinel)
+                         (let [{:keys [completion-promise tenant-conn job-id dataset-id command]} transformation]
+                           (condp = (:type command)
+                             :transformation
+                             (engine/execute-transformation completion-promise
+                                                            tenant-conn
+                                                            job-id
+                                                            dataset-id
+                                                            (:transformation command))
```
Comment (kardan) [inline backend/src/org/akvo/lumen/component/transformation_engine.clj]: Is this line a misstake?

File: backend/src/org/akvo/lumen/transformation/engine.clj [hunk 2]
```diff
@@ -198,95 +198,96 @@
...
-                                     :dataset-version-id new-dataset-version-id
-                                     :dataset-id dataset-id})))
+          source-table (:table-name dataset-version)]
+      (let [{:keys [success? message columns execution-log]} (apply-operation tenant-conn
+                                                                              source-table
+                                                                              columns
+                                                                              transformation)]
+        (if success?
+          (let [new-dataset-version-id (str (util/squuid))]
+            (clear-dataset-version-data-table tenant-conn {:id (:id dataset-version)})
```
Comment (kardan) [inline backend/src/org/akvo/lumen/transformation/engine.clj]: Don't the db calls here need to be in a transaction?

File: backend/src/org/akvo/lumen/component/transformation_engine.clj (hunk 1 above)
Comment (jonase) [inline backend/src/org/akvo/lumen/component/transformation_engine.clj]: Not a mistake, but I can change it if we choose to.

I usually do either

```
(cond
  test-a expr-1
  test-b expr-2)
```

or

```
(cond 
  another-test-1-perhaps-long-and-complicated
  another-expression-also-possibly-a-big-expression

  test-2
  long-and-complicated-expression-for-test-2)
```

Comment (kardan) [inline backend/src/org/akvo/lumen/component/transformation_engine.clj]: I don't mind either but the second arg on new line I find a bit difficult to parse without counting parens. Like:
```
(cond test-a
  test-b)
```

**Pull request merged**